src/pools/balancer/Pool.ts

```diff
@@ -724,19 +728,27 @@ export class Pool {
       new this.web3.eth.Contract(this.poolAbi, poolAddress),
       this.config
     )
+    if (!newPublishMarketSwapFee) {
```

Comment:
see comment above, `newPublishMarketSwapFee` should not be optional and method should throw error when it is not passed